tournament/src/api/views/tournament_views.py

```diff
@@ -81,7 +82,19 @@ def post(request: HttpRequest) -> JsonResponse:
         if json_request.get('registration-deadline') is not None:
             registration_deadline = parser.isoparse(registration_deadline)
             tournament.registration_deadline = TournamentView.convert_to_utc_datetime(registration_deadline)
-        tournament.save()
+        try:
+            tournament.save()
+        except Exception as e:
+            return JsonResponse({'error': str(e)}, status=500)
+
+        try:
+            register_admin_errors = TournamentView.register_admin_as_player(json_request, tournament, user['id'])
+            if register_admin_errors is not None:
+                tournament.delete()
+                return JsonResponse({'errors': register_admin_errors}, status=400)
+        except Exception as e:
+            tournament.delete()
+            return JsonResponse({'errors': [str(e)]}, status=500)
```

Comment:
Maybe `save()` after calling `register_admin_as_player()` so you avoid deleting a tournament just after creating it.

Reply:
I have to save the tournament to be able to link a player to it.